### ISSUE TITLE
Compatibility problems after upgrading programs

### ISSUE THREAD

**Issue by star7th** [casdoor/casdoor#1295]
After  v1.32.0 is upgraded to v1.157.0 （docker image)

1，The super administrator account admin cannot log in with the original password. I don't know if this is related to the encryption method I changed in the background.
2，When I reset the password by email, it prompts "tls: first record does not look like a TLS handshake" before sending the email verification code

**Comment by hsluoyz** [comment]
@star7th 

> 1，The super administrator account admin cannot log in with the original password. I don't know if this is related to the encryption method I changed in the background.

Yes, it's related, don't change it, or you cannot login with original hashed password

> 2，When I reset the password by email, it prompts "tls: first record does not look like a TLS handshake" before sending the email verification code

What Email service do you use?

**Comment by star7th** [comment]
Regarding question 1, I think it should be compatible with the upgrade. When users change the encryption method in the old version, the new version should also read the user configuration to use the new encryption method.
On question 2, use Alibaba Cloud mail service.

**Comment by hsluoyz** [comment]
@star7th for question 2, it's already well supported: https://casdoor.org/docs/provider/email

**Comment by star7th** [comment]
I know you support Alibaba Cloud email service very well. When I use casbin/casdoor: v1.32.0, I can really send emails. This indicates that there is no problem with the configuration. But when I upgraded casbin/casdoor: v1.157.0, an error occurred.

**Comment by hsluoyz** [comment]
@star7th make sure this switch is OFF: https://door.casdoor.com/providers/admin/provider_casbin_email

![image](https://user-images.githubusercontent.com/3787410/201678078-35915586-b480-4cb8-92c0-7cf8045214b0.png)

**Comment by star7th** [comment]
> @star7th make sure this switch is OFF: https://door.casdoor.com/providers/admin/provider_casbin_email
> 
> ![image](https://user-images.githubusercontent.com/3787410/201678078-35915586-b480-4cb8-92c0-7cf8045214b0.png)


When I logged into the management background of v1.32.0, I found that the ssl setting item you mentioned does not exist. So I think this is the option of your new version. However, for the new version, I cannot log in to the management background after upgrading. Naturally, I never set this option.
I checked the provider table of the database. The value of "disable_ssl" is NULL

My feeling is that your upgrade program did not correctly upgrade the database.

**Comment by hsluoyz** [comment]
Hi @star7th can you join our QQ group? https://qm.qq.com/cgi-bin/qm/qr?k=SCBnKNj_1ljeXFT2dk8cwoGQwc5lFy8l&jump_from=webapi

**Comment by star7th** [comment]
Things seem to be getting more and more troublesome.
Let me put this question aside. I directly installed the new version and reconfigured it. No matter the historical data. After all, not many users have been connected before.
As for this problem, I just report that there may be bugs in the historical upgrade program, and you can see if you need to fix them. I build the products I need first.

**Comment by hsluoyz** [comment]
Offline contacted, closed for now
